Classes/common/CDTMutableDocumentRevision.m

```diff
@@ -50,25 +49,14 @@ - (instancetype)initWithDocumentId:(NSString*) documentId
     
     if (self){
         _docId = documentId;
-        _private_body = body ? body : [NSMutableDictionary dictionary];
-        _private_attachments = attachments ? attachments : [NSMutableDictionary dictionary];
+        _body = body ? body : [NSMutableDictionary dictionary];
+        _attachments = attachments ? attachments : [NSMutableDictionary dictionary];
         _sourceRevId = sourceRevId;
     }
     
     return self;
 }
 
-- (void)setBody:(NSDictionary *)body { self.private_body = [body mutableCopy]; }
-
-- (NSMutableDictionary *)body { return self.private_body; }
-
-- (NSMutableDictionary *)attachments { return self.private_attachments; }
-
-- (void)setAttachments:(NSMutableDictionary *)attachments
-{
-    self.private_attachments = [attachments mutableCopy];
```

Comment:
The reason we have the awkward `private_blah` pattern is to allow people to set `NSDictionary` objects for the body and attachments, which we then make mutable copies of. You can see this pain below, with the changes to the tests where setting `body` on mutable document revisions now requires a `mutableCopy` call:

```
rev.body = [@{ @"name" : @"mike", @"age" : @34, @"pet" : @"cat" } mutableCopy];
```

This is a breaking change I'd rather avoid. Is there a way to simplify the Swift developer's code without losing this ability?

For my money, the better fix here is removing the concept of a mutable revision in its entirety, which we've not yet committed to doing.